### PR TITLE
frontend: Configurable OAuth

### PR DESCRIPTION
**Pull request by ivakoleva**
OAuth config was priorly hardcoded in multiple files. We need to be able to produce an image, enabling OAuth configuration upon deployment.

Introduced an appConfig.json asset, and a lean service to read the config upon initialization at first.

Testing done: ci/cd

### CODE REVIEW

File: projects/frontend/data-pipelines/gui/projects/ui/src/app/app-config.service.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    private appConfig: AppConfig;
+
+    constructor(private httpBackend: HttpBackend) {
+        this.httpClient = new HttpClient(httpBackend);
+    }
+
+    loadAppConfig(): Promise<void> {
+        return this.httpClient
+            .get<AppConfig>('/assets/data/appConfig.json')
+            .toPromise()
```
Comment (DeltaMichael) [inline projects/frontend/data-pipelines/gui/projects/ui/src/app/app-config.service.ts:25]: This one is deprecated, please us `firstFrom`.

File: projects/frontend/data-pipelines/gui/projects/ui/src/app/app.module.ts [hunk 2]
```diff
@@ -72,18 +66,45 @@ export function lottiePlayerLoader() {
...
         {
             provide: OAuthStorage,
             useValue: localStorage
         },
         {
             provide: AuthConfig,
-            useValue: authCodeFlowConfig
+            useFactory: (appConfig: AppConfigService) => {
+                return () => {
+                    return appConfig.getAuthCodeFlowConfig();
```
Comment (DeltaMichael) [inline projects/frontend/data-pipelines/gui/projects/ui/src/app/app.module.ts:88]: Nit: No need for all the returns.

```
useFactory: (appConfig: AppConfigService) => () => appConfig.getAuthCodeFlowConfig()
```

File: projects/frontend/data-pipelines/gui/projects/ui/src/app/app.module.ts [hunk 3]
```diff
@@ -72,18 +66,45 @@ export function lottiePlayerLoader() {
     ],
     declarations: [AppComponent, GettingStartedComponent],
     providers: [
+        { provide: AppConfigService, useClass: AppConfigService, deps: [HttpBackend] },
+        {
+            deps: [AppConfigService],
+            multi: true,
+            provide: APP_INITIALIZER,
+            useFactory: (appConfig: AppConfigService) => {
```
Comment (DeltaMichael) [inline projects/frontend/data-pipelines/gui/projects/ui/src/app/app.module.ts:74]: Nit: No need for all the returns

```
useFactory: (appConfig: AppConfigService) => () => appConfig.getAuthCodeFlowConfig()
```

**Review by yonitoo — APPROVED** [review]

File: projects/frontend/data-pipelines/gui/README.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ The gui part of data-pipelines contains:
   * **[Data Pipelines](projects/data-pipelines/README.md)** Library that holds UI Components for managing Data Pipeline API.
   * **[UI](projects/ui/src)** Default UI Wrapper Implementation of the Data Pipeline library
 
-The wrapper ([ui](projects/ui/src)) provides standard OAuth2 authorization flow, configuration of which could be found in [auth.ts](projects/ui/src/app/auth.ts)
+The wrapper ([ui](projects/ui/src)) provides standard OAuth2 authorization flow, configuration of which could be found in [appConfig.json](projects/ui/src/assets/data/appConfig.json)
```
Comment (DeltaMichael) [inline projects/frontend/data-pipelines/gui/README.md:8]: It's not immediately clear how we disable the authorization using the config. Can you please add it to the docs?

Comment (ivakoleva) [reply projects/frontend/data-pipelines/gui/README.md:8]: It's not yet implemented. In the scope of this PR, is making configurable what was pre-existing, without the complexity (yet) of adding new stuff. We will add the additional configs in another PR

Comment (DeltaMichael) [reply projects/frontend/data-pipelines/gui/README.md:8]: Is this scoped anywhere?

**Pull request merged**